### PR TITLE
[WHOAMI] WhoamiGetTokenInfo(): Make it cleaner and safer

### PR DESCRIPTION
**Pull request by SergeGautherie**
JIRA issue: [CORE-11079](https://jira.reactos.org/browse/CORE-11079)

### CODE REVIEW

**Comment by SergeGautherie** [comment]
> Please mention @vicmarcal in the commit message. 

Added, though pretty obvious.

> And the message itself have to be more "clear" about what's done here and why.

Replaced with more details.

File: base/applications/cmdutils/whoami/whoami.c [hunk 1]
```diff
@@ -80,35 +80,43 @@ BOOL WhoamiFree(VOID* Buffer)
...
-                            TokenType,
-                            NULL,
-                            dwLength,
-                            &dwLength);
+        bRet = GetTokenInformation(hToken,
+                                   TokenType,
+                                   NULL,
+                                   0,
+                                   &dwLength);
+        if (bRet || GetLastError() != ERROR_INSUFFICIENT_BUFFER)
```
Comment (learn-more) [inline base/applications/cmdutils/whoami/whoami.c:95]: Did you ever test this code?

**Comment by SergeGautherie** [comment]
> Did you ever test this code?

Tested on LiveCD: results look unchanged.
What do you think is wrong?

**Comment by learn-more** [comment]
If it succeeds you abort?

**Comment by SergeGautherie** [comment]
> If it succeeds you abort?

Yes: as in 98bbe83.

File: base/applications/cmdutils/whoami/whoami.c (hunk 1 above)
Comment (SergeGautherie) [reply base/applications/cmdutils/whoami/whoami.c:95]: Tested on LiveCD: results look unchanged.
What do you think is wrong?

**Comment by SergeGautherie** [comment]
> @Extravert-ir requested changes

Please, update status.

**Comment by GeoB99** [comment]
This is pure bikesheeding -- `NtQueryInformationToken` will always return `STATUS_BUFFER_TOO_SMALL` if you provide a buffer length of 0 (and a NULL pointer). I see no benefits coming form your changes.

**Pull request closed without merging**